### PR TITLE
Added: method to get the main surface

### PR DESCRIPTION
**Pull request by TheNicker**

### CODE REVIEW

**Comment by paroj** [comment]
why is this needed?

**Comment by TheNicker** [comment]
The main reason is because the first surface is not accessible,  the whole separation of the first surface and the subsequent surfaces is wrong in the first place due to the structure of the DDS format, but I guess that's an inherited property from the original project.

So if I have a method like that in my implementation:
` void LoadSurface(const nv_dds::CSurface& surface, Data& myData)`
I have no way to invoke it for the first surface.

File: nv_dds.h [hunk 1]
```diff
@@ -195,6 +195,12 @@ class CDDSImage {
         return m_images[face];
     }
 
+    const CTexture &get_surface() const {
```
Comment (paroj) [inline nv_dds.h]:
```suggestion
    const CTexture &get_surface(unsigned int layer) const {
```

Comment (paroj) [reply nv_dds.h]: while similar to `get_cubemap_face` it would state that you expect an array texture. (with a 2D texture being a special case

Comment (TheNicker) [reply nv_dds.h]: Rebased !
Not really matters to me, I'm not keen on either solution.

File: nv_dds.h [hunk 2]
```diff
@@ -195,6 +195,11 @@ class CDDSImage {
         return m_images[face];
     }
 
+    const CTexture &get_surface(unsigned int layer) const {
+        assert(m_valid);
```
Comment (paroj) [inline nv_dds.h:199]:
```suggestion
        assert(m_valid);
        assert(layer < m_images.size());
```

**Pull request merged**